### PR TITLE
Refactor rank types of celebrations

### LINKED ISSUE #109: Convert Types Array to Dictionary in types.js
Issue #38 was fixed, but if Types was a Dictionary it would improve code readability.

So Types[6] would change into Types.OPT_MEMORIAL.

### PR DESCRIPTION
**Pull request by emagnier**
* Improve code readability for rank types of celebrations.
* eslint: better es6 and Babel support.

resolve #109 

this will help to move forward on #110 

@jarosz, I worked on it last night, and when I opened a PR today I discovered your work [here](https://github.com/jarosz/romcal/commit/92ecc8ba14a99d7cc81ca71bb8fc6d8099bf790f). Our contributions was very close (extend an Array instead of using a Map object, to offer syntactic sugar). And I've include some of your code comments and the way you declare new type properties 👍 (with the `writable: false`). 
Thank you very much for your contribution. I've added you as a co-author of this commit 😉

### CODE REVIEW

**Comment by tukusejssirs** [comment]
I’ll review this in the evening.

However, I’d suggest that we should _always_ assign an issue to ourselves to eliminate duplicate work, and even probably note this in a comment.

**Comment by emagnier** [comment]
Good point @tukusejssirs, we have to use the assignments feature to avoid these situations again.

File: src/calendars/chile.js [hunk 1]
```diff
@@ -52,13 +52,13 @@ let dates = year => {
     // Commented because the key and rank is the same as the definition in general.js
```
Comment (tukusejssirs) [inline src/calendars/chile.js]: I believe the problem lays in the addition of Optional Memorial of St Henry, Bishop of Martyr ([src](https://en.wikipedia.org/wiki/General_Roman_Calendar#Chile)), as on 13 July (when St Henry is celebration in `general.js`) in `chile.js` is Saint Teresa of Los Andes, Virgin (Feast) is celebrated.

I think you should correct this in this PR (and then remove the comment on line 52).

File: src/calendars/poland.js [hunk 2]
```diff
@@ -666,26 +666,26 @@ let dates = year => {
...
       "data": {}
     },
     {
       "key": "saintBarbaraVirginAndMartyr",
-      "type": Types[6],
+      "type": Types.OPT_MEMORIAL,
       "moment": moment.utc({ year: year, month: 11, day: 4 }),
       "data": {}
     },
     // [pejulian] Needs confirmation: In Poland this day is a feast
```
Comment (tukusejssirs) [inline src/calendars/poland.js]: @jarosz, could you confirm this please?

File: src/calendars/chile.js (hunk 1 above)
Comment (emagnier) [reply src/calendars/chile.js]: I hadn't even seen it. What should we do here, remove all this commented block?

Comment (tukusejssirs) [reply src/calendars/chile.js]: You should do this:
1. Remote that comment block.
2. Create a new key (localise it in `en.js`): `"saintCamillusDeLellisPriestOrSaintHenryBishopAndMartyr": "Saint Camillus de Lellis, priest, or Saint Henry, Bishop and Martyr"` (`sk.js`: `Svätého Kamila de Lellis, kňaza, alebo svätého Henricha`).
3. Define `saintCamillusDeLellisPriestOrSaintHenryBishopAndMartyr` celebration in `chile.js`.

Comment (emagnier) [reply src/calendars/chile.js]: Ok, this isn't very related to this PR, but I can do it as a separated commit.

File: src/calendars/poland.js (hunk 2 above)
Comment (tukusejssirs) [reply src/calendars/poland.js]: @emagnier, this one might also be a separate issue then—what do you think?

Comment (emagnier) [reply src/calendars/poland.js]: yes same things

File: src/calendars/chile.js (hunk 1 above)
Comment (tukusejssirs) [reply src/calendars/chile.js]: Moved to issue #116.

File: src/calendars/poland.js (hunk 2 above)
Comment (tukusejssirs) [reply src/calendars/poland.js]: Moved to issue #116.

**Pull request merged**